### PR TITLE
OP-437 Wrong language display in reports

### PR DESCRIPTION
**Pull request by mwithi**
See OP-437.

### CODE REVIEW

File: src/main/java/org/isf/stat/manager/JasperReportsManager.java [hunk 1]
```diff
@@ -347,13 +364,15 @@ public JasperReportResultDto getGenericReportPatientVersion2Pdf(Integer patientI
     public JasperReportResultDto getGenericReportPharmaceuticalOrderPdf(String jasperFileName) throws OHServiceException {
 
         try{
-            Date date = new Date();
+        	HashMap<String, Object> parameters = getHospitalParameters();
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/stat/manager/JasperReportsManager.java:367]: Funny indentation

File: src/main/java/org/isf/stat/manager/JasperReportsManager.java [hunk 2]
```diff
@@ -347,13 +364,15 @@ public JasperReportResultDto getGenericReportPatientVersion2Pdf(Integer patientI
...
+            addBundleParameter(jasperFileName, parameters);
+            
+        	Date date = new Date();
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/stat/manager/JasperReportsManager.java:370]: Here too

File: src/main/java/org/isf/stat/manager/JasperReportsManager.java [hunk 3]
```diff
@@ -522,6 +532,9 @@ public void getGenericReportPharmaceuticalStockCardExcel(String jasperFileName,
     public JasperReportResultDto getGenericReportPharmaceuticalStockWardPdf(Date date, String jasperFileName, Ward ward) throws OHServiceException {
     	
     	try{
+    		HashMap<String, Object> parameters = getHospitalParameters();
+            addBundleParameter(jasperFileName, parameters);
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/stat/manager/JasperReportsManager.java:536]: indentation again; probably the whole file needs to be reformatted

Comment (mwithi) [reply src/main/java/org/isf/stat/manager/JasperReportsManager.java]: As for now I formatted only the blocks indicated. Thanks!

File: src/main/java/org/isf/stat/manager/JasperReportsManager.java [hunk 4]
```diff
@@ -553,6 +623,8 @@ public JasperReportResultDto getGenericReportUserInDatePdf(String fromDate, Stri
 
         try{
             HashMap<String, Object> parameters = compileGenericReportUserInDateParameters(fromDate, toDate, aUser);
+			addBundleParameter(jasperFileName, parameters);
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/stat/manager/JasperReportsManager.java:626]: indentation

File: src/main/java/org/isf/stat/manager/JasperReportsManager.java [hunk 5]
```diff
@@ -773,6 +854,91 @@ public void getGenericReportMYExcel(Integer month, Integer year, String jasperFi
...
+
+	private ResourceBundle getReportResourceBundle(String jasperFileName, String language) {
+		ResourceBundle resourceBundle;
+		try {
+			resourceBundle = ResourceBundle.getBundle(
+					jasperFileName, 
+					new Locale(language), 
+					new UTF8Control());
+		} catch (MissingResourceException e) {
+			logger.error(">> no resource bundle for language = {} found for this report.", language);
```
Comment (dbmalkovsky) [inline src/main/java/org/isf/stat/manager/JasperReportsManager.java:936]: would it make sense to add the jasperFileName to the logging message?

Comment (mizzioisf) [reply src/main/java/org/isf/stat/manager/JasperReportsManager.java:936]: From my perspective yes, it could be useful, @mwithi what do you thing ?

Comment (mwithi) [reply src/main/java/org/isf/stat/manager/JasperReportsManager.java]: yep!

**Comment by mwithi** [comment]
Can you check also the PR https://github.com/informatici/openhospital-gui/pull/444?

**Comment by mwithi** [comment]
@dbmalkovsky and @mizzioisf , as agreed via Slack and Jira, let's merge this issue as is and test it properly on develop before the release

**Pull request merged**